hpccm/primitives/copy.py

```diff
@@ -38,11 +38,18 @@ def __init__(self, **kwargs):
         self.__dest = kwargs.get('dest', '')
         self.__from = kwargs.get('_from', '') # Docker specific
         self.__src = kwargs.get('src', '')
+        self._app = kwargs.get('_app', '') # Singularity specific
 
     def __str__(self):
         """String representation of the primitive"""
         if self.__dest and self.__src:
             if hpccm.config.g_ctype == container_type.DOCKER:
+                if self._app:
```

Comment:
I think Docker should still output instructions in this case?  Behave as if the `_app` attribute was not set.